src/vs/workbench/contrib/terminal/browser/terminalUriParser.ts

```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Schemas } from 'vs/base/common/network';
+import { URI } from 'vs/base/common/uri';
+import { ITerminalIdentifier } from 'vs/workbench/contrib/terminal/browser/terminal';
+
+export function parseTerminalUri(resource: URI): ITerminalIdentifier {
```

Comment:
Call this file `terminalUri.ts` and be a gathering spot for any other URI-related helpers?